source/_cookbook/SSL guide.md

```diff
@@ -0,0 +1,570 @@
+
+Open your browser and go to duckdns.org.
+
+Sign in and create an account using one of the id validation options in the top right corner.
+
+In the domains section pick a name for your subdomain, this can be anything you like, and click add domain.
+
+The URL you will be using later to access your HA instance from outside will be the subdomain you picked, followed by duckdns.org .  For our example we will say our URL is examplehome.duckdns.org
+
+On the top left of duckdns.org select the install option.  Then pick your operating system from the list.  In our example we will use a raspberry pi.  In the dropdown box select the url you just created.
+
+Duckdns.org will now generate personalised instructions for you to follow so that your device can update their website every time your IP address changes.  Carefully follow the instructions given on duckdns.org to set up your device.
+
+At the end of the instructions DuckDNS will suggest you set up port forwarding.  No need, we have already done this in step 2.
+
+What you have now done is set up DuckDNS so that whenever you type examplehome.duckdns.org in to your browser it will convert that to your router's external IP address.  Your external IP address will always be up to date because your device running HA will update DuckDNS every time it changes.
+
+Now type your new URL in to your address bar on your browser with port 8123 on the end:
+
+```text
+http://examplehome.duckdns.org:8123
+```
+
+What now happens behind the scenes is this:
+
+*DuckDNS receives the request and forwards the request to your router's external IP address (which has been kept up to date by your device running HA)
+*Your router receives the request on port 8123 and checks the port forwarding rules
+*It finds the rule you created in step 2 and forwards the request to your HA instance
+*Your browser displays your HA instance frontend.
+
+Did it work? Super!
+
+You now have a remotely accesible HA instance that has a text-based URL and will not drop out if your service provider changes your IP.  But, it is only as secure as the password you set, which can be snooped during your session by a malicious hacker with relative ease.  So we need to set up some encryption with SSL, read on to find out how.
+
+#### {% linkable_title 4 - Obtain an ssl certificate from LetsEncrypt %}
+
+First we need to set up another port forward like we did in step 2.  Set your new rule to:  
+
+```text
+Service name - ha_letsencrypt
+Port Range - 80
+Local IP - YOUR-HA-IP
+Local Port - 80
+Protocol - Both
+```
+
+Remember to save the new rule.
+
+<p class='note'>
+In cases where your ISP blocks port 80 you will need to change the port forward options to forward port 443 from outside to port 443 on your HA device.  Please note that this will limit your options for automatically renewing the certificate, but this is a limitation because of your ISP setup and there is not a lot we can do about it!

```

Comment:
Yeah, this is kinda what I meant in the opening section about not covering every eventuality, and why I said 'novice' rather than noob. If you can think of a better way to word this bit I'm all ears :-)